packages/flutter/test/foundation/caching_iterable_test.dart

```diff
@@ -107,4 +107,19 @@ void main() {
     expect(expanded2, equals(<int>[1, 1, 2, 2, 3, 3, 4, 4, 5, 5]));
     expect(yieldCount, equals(5));
   });
+
+  test('The Caching Iterable: elementAt correctness', () {
+    final Iterable<int> integers = CachingIterable<int>(range(1, 5).iterator);
+    expect(yieldCount, equals(0));
+
+    expect(integers.elementAt(-1), throwsRangeError);
+
+    expect(integers.elementAt(1), equals(2));
+    expect(integers.elementAt(0), equals(1));
+    expect(integers.elementAt(2), equals(3));
+    expect(integers.elementAt(4), equals(5));
+    expect(integers.elementAt(3), equals(4));
+
+    expect(() => integers.elementAt(5), throwsA(isA<RangeError>()));
```

Comment:
Might as well make them match :)

```suggestion
    expect(() => integers.elementAt(5), throwsRangeError);
```